### ISSUE TITLE
Problem.eval_hooks - logits argument

### ISSUE THREAD

**Issue by stefan-falk** [tensorflow/tensor2tensor#1453]
I noticed we can use `Problem.eval_hooks` to run code during evaluation.

```python
def eval_hooks(self, features, logits, hparams):
  # ...
```

However, in doing so, i noticed that the `logits` are usually *almost* what's in `features['targets']` even though, when I quantize the model or test it from checkpoints, the predictions might be far away from the truth.

My question is what exactly `logits` does represent. In `t2t_model.py` I found this:

```python
    # TRAIN and EVAL modes
    if hparams.eval_run_autoregressive and mode == tf.estimator.ModeKeys.EVAL:
      logits, losses_dict = model.eval_autoregressive(features)
    else:
      logits, losses_dict = model(features)  # pylint: disable=not-callable
```

which looks like it should be the raw output from the model - but is that true?

Here is I am making use of `eval_hook` end the input parameters:

```python
class PunctuationHook(tf.train.SessionRunHook):

    def __init__(self, features, logits, hparams):
        self.features = features
        self.logits = logits
        self.hparams = hparams
        vocab_filepath = os.path.join(hparams.data_dir, hparams.problem.vocab_filename)
        self.encdec = SubwordTextEncoder(vocab_filepath)

    def before_run(self, run_context):
        return tf.train.SessionRunArgs((tf.train.get_global_step(), self.features, self.logits))

    def after_run(self, run_context, run_values: SessionRunValues):

        global_step, features, logits = run_values.results
        log_files = glob.glob(os.path.join(self.hparams.model_dir, 'hook.eval-*.log'))
        max_log_files = 3

        if len(log_files) > max_log_files:
            log_files_ = list(map(lambda fp: (int(os.path.basename(fp).split('-')[1].split('.')[0]), fp), log_files))
            to_remove = sorted(log_files_, key=lambda t: -t[0])[max_log_files-1:]
            for _, log_file in to_remove:
                os.remove(log_file)

        log_filepath = os.path.join(self.hparams.model_dir, 'hook.eval-%d.log' % global_step)

        sources = features['inputs'].squeeze().tolist()
        targets = features['targets'].squeeze().tolist()
        predictions = np.argmax(logits, axis=-1).squeeze().tolist()

        with open(log_filepath, 'w') as f:
            for i, (source, target, prediction) in enumerate(zip(sources, targets, predictions)):
                dec_input = self.encdec.decode(source).split('<EOS>')[0].strip()
                dec_target = self.encdec.decode(target).split('<EOS>')[0].strip()
                dec_prediction = self.encdec.decode(prediction[1:]).split('<EOS>')[0].strip()
                f.write('Input :     "%s"\n' % dec_input)
                f.write('Target :    "%s"\n' % dec_target)
                f.write('Prediction: "%s"\n' % dec_prediction)
                f.write('\n')

        with open(log_filepath) as f:
            for i, l in enumerate(f):
                print(l.strip())
                if i >= 100:
                    break
```
